clickhouse-core/src/main/scala/xenon/clickhouse/client/NodeClient.scala

```diff
@@ -34,14 +34,14 @@ object NodeClient {
 class NodeClient(val nodeSpec: NodeSpec) extends AutoCloseable with Logging {
 
   private val node: ClickHouseNode = ClickHouseNode.builder()
+    .options(nodeSpec.options)
     .host(nodeSpec.host)
     .port(nodeSpec.protocol, nodeSpec.port)
     .database(nodeSpec.database)
     .credentials(ClickHouseCredentials.fromUserAndPassword(nodeSpec.username, nodeSpec.password))
     .build()
 
   private val client: ClickHouseClient = ClickHouseClient.builder()
-    .option(ClickHouseClientOption.ASYNC, false)
```

Comment:
did you mean to remove it?

Reply:
User can setting it by `spark.sql.catalog.<catalog_name>.option.async`

I'm still thinking how to make the option overwriting logic meaningful, you can compile and test if this approach works first.